# The AGG_paradox

# Understanding the association of AGG codons with highly expressed genes, an artifact or an enigma? 

This project begins by using data from Goodman et al., (Causes and effects of N-terminal codon bias in bacterial genes 2013). This group took 137 endogenous E.coli genes and created over 14,000 constructs through by using two different reporters, 3 different ribosomal binding sites and 13 codon variants for the first 11 amino acids (including initiation Met) to understand how 5â€™ codons affect protein expression. In their data 

![Image Description](https://github.com/user-attachments/assets/89f121df-bf7e-4d86-be99-b4d23137524a)


<img width="640" alt="image" src="https://github.com/user-attachments/assets/89f121df-bf7e-4d86-be99-b4d23137524a" />


Within the paper, they presented their expression data in the form of log2 fold change which Dr Hurst finds strange. On top of this, in their CSV file they have three different metrics of how much expression occurs when measuring GFP as a reporter. I will look to use these metrics and hopefully other data to understand if the AGG paradox seen in the Goodman paper is something worth investigating. 
